backend/internal/userschema/service/userschemaservice.go

```diff
@@ -488,3 +492,394 @@ func logAndReturnServerError(
 	logger.Error(message, log.Error(err))
 	return &constants.ErrorInternalServerError
 }
+
+// ValidateUser validates user attributes against the user schema for the given user type.
+func (us *UserSchemaService) ValidateUser(userType string, userAttributes json.RawMessage) *serviceerror.ServiceError {
+	logger := log.GetLogger().With(log.String(log.LoggerKeyComponentName, userSchemaLoggerComponentName))
+
+	if userType == "" {
+		logger.Debug("User type is empty, skipping schema validation")
+		return nil
+	}
+
+	if userAttributes == nil {
+		logger.Debug("User has no attributes to validate")
+		return nil
+	}
+
+	if len(userAttributes) == 0 {
+		logger.Debug("User has no attributes to validate")
+		return nil
+	}
+
+	userSchema, err := store.GetUserSchemaByName(userType)
+	if err != nil {
+		if errors.Is(err, constants.ErrUserSchemaNotFound) {
+			logger.Debug("No schema found for user type, skipping validation", log.String("userType", userType))
+			return nil // Allow users without schema
+		}
+		return logAndReturnServerError(logger, "Failed to get user schema", err)
+	}
+
+	if err := validateUserAttributesAgainstSchema(userAttributes, userSchema.Schema); err != nil {
+		logger.Debug("Schema validation failed", log.Error(err), log.String("userType", userType))
+		return &constants.ErrorUserValidationFailed
```

Comment:
Instead of a service error, how about returning a boolean indicating the validation status? We can return a boolean and Service error from the function; service error for other error scenarios

Reply:
Fixed with https://github.com/asgardeo/thunder/compare/ac64c35f600950c4c2150205e3d8e693faef7603..50d21a542f55a15f2b3d369a94a7278220574656